src/servers/middlewares/wallet-id.ts

```diff
@@ -16,12 +16,12 @@ const validateWalletId = async (
   if (!walletId) return new Error("Invalid wallet")
   if (walletId instanceof Error) return walletId
 
-  if (!context.domainUser) {
-    return new Error("Invalid wallet")
+  if (!context.domainAccount) {
+    return new Error("Invalid Account")
   }
 
   const hasPermissions = await Accounts.hasPermissions(
-    context.domainUser.id,
+    context.domainAccount.id,
```

Comment:
since this is a top level use case we could also pass the whole `domainAccount` instead of `id`. I know out of scope for this PR - just thinking.